lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2632,7 +2632,13 @@ buildAndSignTransactionNew
     balanceTx protocolParams utxo unsignedTxBody = do
         let protocolParameters =
                 ( protocolParams
-                , fromJust $ currentNodeProtocolParameters protocolParams
+                , fromMaybe
+                    (error $ unwords
+                        [ "buildAndSignTransactionNew: no nodePParams."
+                        , "should only be possible in Byron, where"
+                        , "withRecentEra should prevent this to be reached."
+                        ])
+                    $ currentNodeProtocolParameters protocolParams
```

Comment:
🥇

Reply:
2nd thought: this unsafe `fromJust` is used in more that one place, so it makes sense to use your fix in all other places too.